### PR TITLE
 Canvas2D: Animation and ActionManager support + bugs fixes

### PR DESCRIPTION
**Pull request by nockawa**
ActionEvent: adding a CreateNewFromPrimitive (which takes a prim as 'any' to allow building Core)

SmartPropertyPrim:
 - Adding the animation proeprty to support animations.
 - Adding Inline documentation

Prim2DBase:
 - Adding actionManager get property, supporting every trigger except KeyDown/Up and IntersectionEnter/Exit

Text2D: improved rendering quality and fix Origin computation

FontTexture: Now use NEAREST sampling mode to achieve a better rendering quality

### CODE REVIEW

File: src/Animations/babylon.animation.ts [hunk 1]
```diff
@@ -453,15 +468,16 @@
                 destination = this._target;
             }
 
+            if (!this._originalBlendValue) {
```
Comment (deltakosh) [inline src/Animations/babylon.animation.ts]: why did you move this outside the loop?

File: src/Animations/babylon.animation.ts [hunk 2]
```diff
@@ -482,7 +498,7 @@
             }
 
             if (this._target.markAsDirty) {
-                this._target.markAsDirty(this.targetProperty);
+                this._target.markAsDirty(this.targetProperty, this._originalBlendValue);
```
Comment (deltakosh) [inline src/Animations/babylon.animation.ts]: Is this mandatory? actions are not providing the 2nd parameter for now

Comment (nockawa) [inline src/Animations/babylon.animation.ts]: I figured that in JS it's no big deal to call with a second arg, right?

Comment (deltakosh) [inline src/Animations/babylon.animation.ts]: My point is: actions will NOT provide the 2nd parameter.
